apps/crn-server/src/data-providers/reminders.data-provider.ts

```diff
@@ -214,6 +264,81 @@ const getResearchOutputRemindersFromQuery = (
   );
 };
 
+const getResearchOutputDraftRemindersFromQuery = (
+  queryResearchOutputsContents: FetchReminderDataQuery['draftResearchOutputs'],
+  findUsersContent: FetchReminderDataQuery['findUsersContent'],
+): ResearchOutputDraftReminder[] => {
+  if (!findUsersContent || !findUsersContent.flatData.teams) {
+    return [];
+  }
+
+  const userTeamIds = getUserTeamIds(findUsersContent.flatData.teams);
+  const userWorkingGroupIds = getUserWorkingGroupIds(
+    findUsersContent.referencingWorkingGroupsContents,
+  );
+
+  if (!queryResearchOutputsContents) {
+    return [];
+  }
+
+  return queryResearchOutputsContents.reduce<ResearchOutputDraftReminder[]>(
+    (researchOutputReminders, researchOutput) => {
+      const userName = getUserName(researchOutput);
+      const { associationName, associationType } =
+        getAssociationNameAndType(researchOutput);
+
+      if (
+        !researchOutput.flatData.teams ||
+        !researchOutput.flatData.title ||
+        userName === null ||
+        associationName === null ||
+        associationType === null
+      ) {
+        return researchOutputReminders;
+      }
+
+      const researchOutputTeams = researchOutput.flatData.teams.map(
+        (team) => team.id,
+      );
+      const researchOutputWorkingGroups = (
+        researchOutput.flatData.workingGroups || []
+      ).map((workingGroup) => workingGroup.id);
+
+      const isInTeam = researchOutputTeams.some((team) =>
+        userTeamIds.includes(team),
+      );
+
+      const isInWorkingGroup = researchOutputWorkingGroups.some(
+        (workingGroup) => userWorkingGroupIds.includes(workingGroup),
+      );
+
+      if (
+        (associationType === 'team' && !isInTeam) ||
+        (associationType === 'working group' && !isInWorkingGroup)
+      ) {
+        return researchOutputReminders;
+      }
```

Comment:
hey, reading the ticket again, I've noticed this "**incl. ASAP Hub role=Staff**"

> As a member of the team that is associated with the draft output (PM or Non-PM; **incl. ASAP Hub role=Staff**)

I would ask Tiff before implementing it, but if this requirement above is needed, I should see the reminder even if I don't belong to the working group or team, but I have Asap Role = Staff, so the logic here would change a little bit. Sorry, just noticed this now 😅

Reply:
Done, ty for noticing